client/me/purchases/manage-purchase/change-payment-method/index.jsx

```diff
@@ -169,20 +149,54 @@ ChangePaymentMethod.propTypes = {
 	isFullWidth: PropTypes.bool.isRequired,
 };
 
+/**
+ * @param {object} payment Payment object taken from the purchase
+ * @param {string=} payment.type Payment type; description exists only to satisfy eslint
```

Comment:
We started messing with this and got distracted. I think the `null|undefined` makes sense though there's also a `creditCard: { id: unknown; }` if `type` is `'credit_card'`.

Reply:
Heh, this docblock is really only there to document that the output should be a payment method ID as used by `CheckoutPaymentMethods`. The `payment` object comes from the endpoint and I'm not convinced it's worth the effort to try to type it since the endpoint is not principled anyway.

Reply:
Properly typing this doesn't even provide very good type safety within the function, so it doesn't seem worth the time.